lib/examples/ext_ctrls_test/main.rs

```diff
@@ -0,0 +1,29 @@
+use nix::fcntl::{open, OFlag};
+use nix::sys::stat::Mode;
+use std::fs::File;
+use std::os::unix::io::FromRawFd;
+use v4l2r::ioctl::*;
+
+const DEVICE_PATH: &str = "/dev/video2";
+
+fn main() {
+    let fd = unsafe {
+        File::from_raw_fd(
+            open(DEVICE_PATH, OFlag::O_RDWR | OFlag::O_CLOEXEC, Mode::empty())
+                .expect(std::format!("Cannot open {}", DEVICE_PATH).as_str()),
+        )
+    };
+
+    match g_ext_ctrl(&fd, None, ExtControlKind::FwhtParams) {
+        Ok(fwht_params) => {
```

Comment:
You can directly match `Ok(ExtControl::FwhtParams(fwht_params))` here and avoid the `if let` statement below. If that `g_ext_ctrl` call returned a control different from `FwhtParams`, this would be an error anyway and we wouldn't want to proceed.

Reply:
I can. But then: 1) I need to add handling of variants other than ExtControl::FwhtParams (1 line for an additional match arm) and 2) I need to instantiate an ExtControl::FwhtParams() before calling s_ext_ctrl(). This way I exchange 4 lines of code for another 4 lines of code. I still can do it if it is a cleaner approach. What do you think?